src/components/Error/ErrorBoundary.tsx

```diff
@@ -28,19 +32,52 @@ export default class ErrorBoundary extends Component<PropsWithChildren<ErrorBoun
     this.props.onError?.(error, errorInfo)
   }
 
+  getErrorDialogHeader(error: Error): string {
+    if (error instanceof MetaMaskConnectionError) {
+      return t`Wallet disconnected`
+    }
+    return t`Please refresh the page and try again.`
+  }
+
+  getErrorDialogActionLabel(error: Error): string {
+    if (error instanceof MetaMaskConnectionError) {
+      return t`Reload`
+    }
+    return t`Reload the page`
+  }
+
+  getErrorDialog(error: Error) {
+    return (
+      <Dialog color="dialog">
+        <ErrorDialog
+          error={error}
+          header={this.getErrorDialogHeader(error)}
+          action={this.getErrorDialogActionLabel(error)}
+          onClick={() => window.location.reload()}
+        />
+      </Dialog>
+    )
+  }
+
   render() {
-    if (this.state.error) {
-      return (
-        <Dialog color="dialog">
-          <ErrorDialog
-            error={this.state.error}
-            header={<Trans>Please refresh the page and try again.</Trans>}
-            action={<Trans>Reload the page</Trans>}
-            onClick={() => window.location.reload()}
-          />
-        </Dialog>
-      )
+    if (this.props.error != null || this.state.error != null) {
+      return this.getErrorDialog(this.props.error ?? (this.state.error as Error))
     }
     return this.props.children
   }
 }
+
+/**
+ * This is a wrapper around the ErrorBoundary class component above.
+ * The class component handles regular Errors "thrown" during render.
+ *
+ * For other errors, which rely on reading from state, we need this functional component.
+ */
+export default function ErrorBoundary({ children, onError }: PropsWithChildren<ErrorBoundaryProps>) {
```

Comment:
I don't think this needs a wrapper. The context can be checked in the `render` function.